modules/standard/FileSystem.chpl

```diff
@@ -574,19 +571,34 @@ proc locale.cwd(out error: syserr): string {
    :rtype: `string`
 */
 proc locale.cwd(): string throws {
+  extern proc chpl_fs_cwd(ref working_dir:c_string):syserr;
+
+  var ret:string;
   var err: syserr = ENOERR;
-  var ret = cwd(err);
+  on this {
+    var tmp:c_string;
+    // c_strings can't cross on statements.
+    err = chpl_fs_cwd(tmp);
+    ret = if err
+          then ""
+          else new string(tmp, needToCopy=false);
```

Comment:
Can this just be `ret = tmp:string`? I don't like the call to the string constructor here.